ui/helpers/utils/util.js

```diff
@@ -750,3 +750,16 @@ export const hexToText = (hex) => {
 export const getAvatarFallbackLetter = (subjectName) => {
   return subjectName?.match(/[a-z0-9]/iu)?.[0] ?? '?';
 };
+
+/**
+ * Get checksum address for the address string passed.
+ *
+ * @param {string} address - The address passed.
+ * @returns Checksum address or undefined.
+ */
+export const getChecksumAddress = (address) => {
+  if (!ethUtil.isValidAddress(address)) {
+    return undefined;
```

Comment:
Was the intent of this fix and util not to convert the integer address to hexadecimal rather than just displaying the original value?

Reply:
we still want to display invalid address to user with error from blockaid to show its wrong

Reply:
The `Name` component is designed to show Ethereum addresses as hexadecimal strings, not whatever unformatted input is given to it.

We want the UX to be consistent, so it seems the best two choices are to reject the request outright if an address is in a bad format, or normalize it so we can render it in a consistent format, plus this would ensure Blockaid can validate it correctly rather than throwing.

Reply:
For a wrong address, I think we should show users that address value as it is so its more obvious to user's what is wrong and not change that.

Reply:
Let's discuss this more with product team to reach on decision here.